lib/sqlalchemy/sql/base.py

```diff
@@ -1283,7 +1308,12 @@ def _execute_on_scalar(
         ) -> Any: ...
 
     @util.ro_non_memoized_property
-    def _all_selected_columns(self):
+    def _all_selected_columns(
+        self,
+    ) -> Union[
```

Comment:
we seem to using `_SelectIterable` in many subclasses. I think it would be ok here too